sky/templates/aws-ray.yml.j2

```diff
@@ -121,13 +121,13 @@ head_start_ray_commands:
   # Line "which prlimit ..": increase the limit of the number of open files for the raylet process, as the `ulimit` may not take effect at this point, because it requires
   # all the sessions to be reloaded. This is a workaround.
   - ((ps aux | grep "-m sky.skylet.skylet" | grep -q python3) || nohup python3 -m sky.skylet.skylet >> ~/.sky/skylet.log 2>&1 &);
-    ray stop; ray start --disable-usage-stats --head --port=6379 --object-manager-port=8076 --autoscaling-config=~/ray_bootstrap_config.yaml {{"--resources='%s'" % custom_resources if custom_resources}};
-    which prlimit && for id in $(pgrep -f raylet/raylet); do sudo prlimit --nofile=1048576:1048576 --pid=$id; done;
+    ray stop; ray start --disable-usage-stats --head --port=6379 --object-manager-port=8076 --autoscaling-config=~/ray_bootstrap_config.yaml {{"--resources='%s'" % custom_resources if custom_resources}} --num-gpus={{num_gpus}} || exit 1;
```

Comment:
No, ray has different meaning for `--resources` and `--num-gpus`, and it will handle the `num-gpus` separately. We have to specify the `num-gpus` to make sure our `inferentia` task can be correctly scheduled.

Reply:
Just to confirm: I tried to `sky launch` an Inferentia VM and from the ray status we already have `0.0/1.0 Inferentia`. do we still need to manually add `--num-gpus`?
```
ubuntu@ip-172-31-18-99:~$ ray status
======== Autoscaler status: 2022-10-18 05:09:33.061513 ========
Node status
---------------------------------------------------------------
Healthy:
 1 ray.head.default
Pending:
 (no pending nodes)
Recent failures:
 (no failures)

Resources
---------------------------------------------------------------
Usage:
 0.0/4.0 CPU
 0.0/1.0 GPU
 0.0/1.0 Inferentia
 0.00/3.939 GiB memory
 0.00/1.969 GiB object_store_memory

Demands:
 (no resource demands)
```

Reply:
Ahh, actually, it is possible to remove the `num-gpus`. I fixed our job submission logic, so that we can remove it. Thanks for the suggestions @infwinston @michaelzhiluo !